### ISSUE TITLE
Migration results in unusable csproj

### ISSUE THREAD

**Issue by MattGertz** [dotnet/cli#5341]
## Steps to reproduce
Migration this project.json:
```
{
  "version": "1.0.0-*",
  "buildOptions": {
    "emitEntryPoint": true, 
    "debugType": "portable"
    },

  "dependencies": {
    "Microsoft.NETCore.App": {
      "type": "platform",
      "version": "1.0.0-rc2-24018"
    }
  },
  "frameworks": {
    "netstandard1.5": {}
  }
}
```
## Expected  behavior
Should be something like:
```
<?xml version="1.0" encoding="utf-16"?>
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netcoreapp1.0</TargetFramework>
    <DebugType>portable</DebugType>
    <AssemblyName>vigenere</AssemblyName>
    <OutputType>Exe</OutputType>
    <PackageId>vigenere</PackageId>
  </PropertyGroup>
  <ItemGroup>
  </ItemGroup>
</Project>
```
## Actual behavior
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netstandard1.5</TargetFramework>
    <DebugType>portable</DebugType>
    <AssemblyName>vigenere - original - Copy</AssemblyName>
    <OutputType>Exe</OutputType>
    <PackageId>vigenere - original - Copy</PackageId>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.NETCore.App" Version="1.0.3" />
  </ItemGroup>

</Project>

However, netstandard1.5 and M.NC.A 1.0.3 are mutually incompatible.  

If you open this in VS, you'll get an error to this effect, but the manual fix (removing the package reference or downgrading netstandard to 1.0.4) is non-obvious.

## Environment data
`dotnet --info` output:
.NET Command Line Tools (1.0.0-rc3-004504)

Product Information:
 Version:            1.0.0-rc3-004504
 Commit SHA-1 hash:  b312fbb176

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x86
 Base Path:   C:\Program Files (x86)\dotnet\sdk\1.0.0-rc3-004504

**Comment by eerhardt** [comment]
@MattGertz -

The project.json has 

```
 "frameworks": {
    "netstandard1.5": {}
  }
```

but in the expected .csproj you list:

```
<TargetFramework>netcoreapp1.0</TargetFramework>
```

How should `dotnet migrate` know to do this mapping?  Does the original project.json work fully using a "preview2" CLI?  Does it work correctly if the project.json lists `netcoreapp1.0` instead?

**Comment by MattGertz** [comment]
According to Livar, the E2E used to be able to cope with this, and we wanted to talk about it, so filing this issue to do exactly that.  (FWIW, I had no problems running this app at the time.)

**Comment by livarcocc** [comment]
@eerhardt @MattGertz I believe I understand this now. M.NC.App 1.0.0-rc2, which Matt was using before, was compatible with netstandard1.5. However, once we moved to 1.0.0, that is no longer the case.

Because we decided to update all packages to LTS, now that package is updated to 1.0.3, which is no longer compatible with netstandard1.5.

I am not sure what to do here. I am reticent to make migration understand the package compatibility graph and update projects to that their packages continue to be compatible.
